libheif/heif_context.cc

```diff
@@ -1847,6 +1842,10 @@ Error HeifContext::Image::encode_image_as_hevc(std::shared_ptr<HeifPixelImage> i
 
     std::shared_ptr<HeifPixelImage> alpha_image;
     alpha_image = create_alpha_image_from_image_alpha_channel(image);
+    auto nclx = std::make_shared<color_profile_nclx>();
+    nclx->set_default();
+    nclx->set_full_range_flag(true); // in default, but just to be sure in case defaults change
+    alpha_image->set_color_profile_nclx(nclx);
```

Comment:
I am not familiar with HEIC, so I preserved the current behavior in the encode_image_as_hevc() method and only changed the behavior in the encode_image_as_av1() method.

Please feel free to edit this pull request or replace it with your own as you see fit. Thank you!